homeassistant/components/rest_command/__init__.py

```diff
@@ -159,18 +167,44 @@ async def async_service_handler(service: ServiceCall) -> None:
                             payload,
                         )
 
-            except asyncio.TimeoutError:
+                    if service.return_response:
+                        _content = None
+                        if response.content_type == "application/json":
+                            try:
+                                _content = await response.json()
+                            except (JSONDecodeError, AttributeError) as err:
+                                raise HomeAssistantError from err
+                        else:
+                            try:
+                                _content = await response.text()
+                            except (LookupError, UnicodeDecodeError) as err:
```

Comment:
It's been a while since writing this and at the moment I cannot reproduce. I believe that for empty responses this could be raised which should indicate that indeed for JSON it should be applicable too.
With the restructuring in order to get the McCabe complexity to a passable level this error would also be caught for JSON responses.

Reply:
I've never seen LookupError caught when parsing an aiohttp text response so this seems odd to me -- i'd say remove unless we get more conviction its actually thrown.